src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -28,14 +32,46 @@
      * 26.04.2019 Сергей 7 100
      * 26.04.2019 София 9 100
      * 26.04.2019 Сергей 11 50
-     *
+     * <p>
      * Пример вывода:
      * Отчёт за период 01.04.2019  - 30.04.2019
      * Сергей - 1550
      * Андрей - 600
      * София - 900
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy", Locale.ENGLISH);
+
+        try {
+            if (dateFormat.parse(dateFrom).after(dateFormat.parse(dateTo))) {
+                return null;
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        String infoString = "";
+        infoString += "Отчёт за период " + dateFrom + " - " + dateTo + "\n";
+        for (int i = 0; i < names.length; i++) {
+            int currentWorkerSalary = 0;
+            for (String nextData : data) {
+                String[] parsedData = nextData.split(" ");
+                try {
+                    if (dateFormat.parse(parsedData[0]).after(dateFormat.parse(dateTo))) {
```

Comment:
@NikolayKatsai you can move the date parsing to a separate private method. Then the code will be cleaner